server.go

```diff
@@ -875,26 +875,51 @@ func (s *Server) ReadMetricSocket(serverConn net.PacketConn, packetPool *sync.Po
 			log.WithError(err).Error("Error reading from UDP metrics socket")
 			continue
 		}
-		if n > s.metricMaxLength {
-			metrics.ReportOne(s.TraceClient, ssf.Count("packet.error_total", 1, map[string]string{"packet_type": "unknown", "reason": "toolong"}))
-			continue
-		}
+		s.processMetricPacket(n, buf, packetPool)
+	}
+}
+
+// Splits the read metric packet into multiple metrics and handles them
+func (s *Server) processMetricPacket(numBytes int, buf []byte, packetPool *sync.Pool) {
+	if numBytes > s.metricMaxLength {
+		metrics.ReportOne(s.TraceClient, ssf.Count("packet.error_total", 1, map[string]string{"packet_type": "unknown", "reason": "toolong"}))
+		return
+	}
+
+	// statsd allows multiple packets to be joined by newlines and sent as
+	// one larger packet
+	// note that spurious newlines are not allowed in this format, it has
+	// to be exactly one newline between each packet, with no leading or
+	// trailing newlines
+	splitPacket := samplers.NewSplitBytes(buf[:numBytes], '\n')
+	for splitPacket.Next() {
+		go s.HandleMetricPacket(splitPacket.Chunk())
```

Comment:
I think this should remain serial for now also!

Reply:
My bad, sorry missed , fixed it now